src/opensles/AudioStreamOpenSLES.cpp

```diff
@@ -248,3 +250,29 @@ int64_t AudioStreamOpenSLES::getFramesProcessedByServer() const {
     int64_t framesProcessed = millis64 * getSampleRate() / kMillisPerSecond;
     return framesProcessed;
 }
+
+Result AudioStreamOpenSLES::waitForStateChange(StreamState currentState,
+                                                     StreamState *nextState,
+                                                     int64_t timeoutNanoseconds) {
+    LOGD("AudioStreamOpenSLES::waitForStateChange()");
+
+    if (getState() == StreamState::Closed) return Result::ErrorClosed;
```

Comment:
Maybe does not need to be an error. Just let it run.

Reply:
done

Reply:
and test updated to test for `Result::OK` when stream is closed and `waitForStateChange` is called.